src/trans.rs

```diff
@@ -250,20 +322,78 @@ impl<'v, 'tcx: 'v> BinaryenFnCtxt<'v, 'tcx> {
 
                         match *destination {
                             Some((ref lvalue, _)) => {
-                                let (lvalue_index, _) = self.trans_lval(lvalue);
+                                let dest = self.trans_lval(lvalue);
 
                                 if b_fnty == BinaryenNone() {
                                     // The result of the Rust call is put in MIR into a tmp local,
                                     // but the wasm function returns void (like the print externs)
                                     // TODO: what representation should the unit type have in wasm ?
-                                    debug!("emitting {:?} Call to fn {:?} + SetLocal for unit type", call_kind, func);
+                                    debug!("emitting {:?} Call to fn {:?} + SetLocal({}) for unit type", call_kind, func, dest.index.0);
                                     binaryen_stmts.push(b_call);
 
                                     let unit_type = BinaryenConst(self.module, BinaryenLiteralInt32(-1));
```

Comment:
and by "exactly" I actually mean not skipping all those lines but just the set_local, we'll still need to call the fn ;)